lib/images/ImageLoader/index.js

```diff
@@ -0,0 +1,98 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { Icon } from '../..';
+
+class ImageLoader extends PureComponent {
+  static propTypes = {
+    alt: PropTypes.string.isRequired,
+    src: PropTypes.string.isRequired,
+    setLoadedDimensions: PropTypes.bool,
+    preLoadStyles: PropTypes.shape(),
+    onLoad: PropTypes.func,
+    onError: PropTypes.func
+  };
+
+  static defaultProps = {
+    setLoadedDimensions: false,
+    preLoadStyles: {},
+    onLoad: () => {},
+    onError: () => {}
+  };
+
+  state = {
+    imageHasLoaded: false,
+    showImage: false,
+    imageHasErrored: false
+  };
+
+  showImage = () =>
+    window.requestAnimationFrame(() => this.setState({ showImage: true }));
```

Comment:
What does `window.requestAnimationFrame` do?

Reply:
Basically adds a timeout so we get a transition between the loaded and show state. This is required so when we animate the dimensions between the state we get some time and the browser can step through them automatically.

I didn't want to add a setTimeout because the time I would use would be bullshit.